### PR TITLE
fix(tracking): fix setting of remaining effort

### PR DESCRIPTION
**Pull request by trowik**
Change signal to pre_save and update most_recent_remaining_effort on task and total_remaining_effort on project if the value of remaining_effort on report has changed.

### CODE REVIEW

File: timed/tracking/signals.py [hunk 1]
```diff
@@ -15,20 +15,28 @@ def update_rejected_on_reports(sender, instance, **kwargs):
...
-            .get("sum_remaining")
-        )
-        project.total_remaining_effort = total_remaining_effort
-        project.save()
+    # ignore signal when loading a fixture
+    if kwargs.get("raw", False):  # pragma: no cover
+        return
+
+    if instance.pk:
+        if instance.task.project.remaining_effort_tracking:
```
Comment (fugal-dy) [inline timed/tracking/signals.py:26]: It's more agreeable when reading code if unmet conditions are ruled out immediately aka free up memory where possible and stay a close to the left border as possible.

```suggestion
    if not instance.pk:
        return
    if instance.task.project.remaining_effort_tracking is not True:
        return
```

File: timed/tracking/tests/test_report.py [hunk 2]
```diff
@@ -1954,3 +1954,71 @@ def test_report_remaining_effort_total(
...
+
+    url = reverse("report-detail", args=[report_2.id])
+
+    response = internal_employee_client.patch(url, data)
+    assert response.status_code == status.HTTP_200_OK
+
+    report.task.refresh_from_db()
+    assert report.task.most_recent_remaining_effort == timedelta(hours=3)
+    assert report.task.project.total_remaining_effort == timedelta(hours=3)
+
```
Comment (fugal-dy) [inline timed/tracking/tests/test_report.py:2006]: maybe some comment along the line that the most recent remaining effort value should only be updated if and only if the `remaining_effort` property has changed on the report instance (and not blindly take the latest report that might have changed something else completely)

**Review by fugal-dy — APPROVED** [review]
good job =)

File: timed/tracking/signals.py (hunk 1 above)
Comment (trowik) [reply timed/tracking/signals.py]: makes sense, thanks :)

File: timed/tracking/tests/test_report.py (hunk 2 above)
Comment (trowik) [reply timed/tracking/tests/test_report.py:2006]: Updated the doc string in the signal

**Pull request merged**